lib/remotable/nosync.rb

```diff
@@ -1,37 +1,84 @@
+require "active_resource/threadsafe_attributes"
+
 module Remotable
   module Nosync
 
-
-    def nosync!
-      self.nosync = true
+    def self.included(base)
+      base.include InstanceMethods
     end
 
-    def nosync(new_value=true)
-      old_value = _nosync
-      self.nosync = new_value
-      yield
-    ensure
-      @nosync = old_value
+    def self.extended(base)
+      base.extend ClassMethods
     end
 
-    def nosync=(val)
-      @nosync = val
-    end
+    module InstanceMethods
 
-    def nosync_value?
-      !_nosync.nil?
-    end
+      def nosync!
+        self.nosync = true
+      end
 
-    def nosync?
-      @nosync == true
-    end
+      def reset_nosync!
+        remove_instance_variable(:@nosync) if instance_variable_defined?(:@nosync)
+      end
+
+      def nosync(new_value=true)
+        old_value = _nosync
+        self.nosync = new_value
+        yield
+      ensure
+        @nosync = old_value
+      end
+
+      def nosync=(val)
+        @nosync = val
+      end
+
+      def nosync_value?
+        !_nosync.nil?
+      end
+
+      def nosync?
+        _nosync == true
```

Comment:
⬇️ The `== true` part is probably not necessary, but I'm also ok with only making changes necessary to the current code red issue 😅

Reply:
Yeah, I think we should be okay if we want to just refactor this to not use `== true`.